integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java

```diff
@@ -103,136 +129,276 @@
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_RETRYING,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_RETRYING, "Normal", timestamp));
+  /**
+   * Test verifies there is only 1 DomainProcessing Starting/Completed event is logged in a multi cluster domain.
+   */
+  @Order(5)
+  @Test
+  public void testK8SEventsMultiClusterEvents() {
+    createNewCluster();
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    scaleClusterWithRestApi(domainUid, cluster2Name, 1,
+        externalRestHttpsPort, opNamespace, opServiceAccount);
+    logger.info("verify the DomainProcessing Starting/Completed event is generated");
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_PROCESSING_STARTING, "Normal", timestamp);
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_PROCESSING_COMPLETED, "Normal", timestamp);
+    logger.info("verify the only 1 DomainProcessing Starting/Completed event is generated");
+    assertEquals(1, getEventCount(domainNamespace1, domainUid, DOMAIN_PROCESSING_STARTING, timestamp));
+    assertEquals(1, getEventCount(domainNamespace1, domainUid, DOMAIN_PROCESSING_COMPLETED, timestamp));
+  }
 
-    //verify domain processing aborted event
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_ABORTED,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_ABORTED, "Warning", timestamp));
+  /**
+   * Scale the cluster beyond maximum dynamic cluster size and verify the warning event is generated.
+   */
+  @Order(6)
+  @Test
+  @Disabled("Bug - OWLS-87019")
+  public void testDomainK8sEventsScalePastMax() {
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    scaleClusterWithRestApi(domainUid, cluster1Name, 3, externalRestHttpsPort, opNamespace, opServiceAccount);
+    logger.info("verify the DomainValidationError event is generated");
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_VALIDATION_ERROR, "Warning", timestamp);
+  }
 
-    timestamp = new DateTime(System.currentTimeMillis() - 2000);
+  /**
+   * Scale the cluster below minimum dynamic cluster size and verify the warning event is generated.
+   */
+  @Order(7)
+  @Test
+  @Disabled("Bug - OWLS-87019")
+  public void testDomainK8sEventsScaleBelowMin() {
```

Comment:
do you have `allowReplicasBelowMinDynClusterSize` set to false in the cluster?

Reply:
after setting this to false I can see the event being logged.